File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        for (int i = 0; i < nums.size(); i++) {
            int remainigTarget = target - nums[i];
            if (umap[remainigTarget]) {
                return {i, umap[remainigTarget] - 1};
            }
            umap[nums[i]] = i + 1;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const map = new Map();
    const ans = []
    for (let i = 0; i < nums.length; i++) {
        const newTarget = target - nums[i];
        if (map.has(newTarget)) {
            return [map.get(newTarget), i];
        }
        else {
            map.set(nums[i], i)
        }
    }
    return []
};